File: apps/patter-voice-sdk/libraries/typescript/src/stt/speechmatics.ts
/** Speechmatics streaming STT for Patter pipeline mode. */
import {
  SpeechmaticsSTT as _SpeechmaticsSTT,
  type SpeechmaticsSTTOptions as _SpeechmaticsSTTOptions,
} from "../providers/speechmatics-stt";

// Re-export the canonical options type from the provider so callers have a
// single source of truth and can import either path.
export type SpeechmaticsSTTOptions = _SpeechmaticsSTTOptions & {
  /** API key. Falls back to SPEECHMATICS_API_KEY env var when omitted. */
  apiKey?: string;
};

/**
 * Speechmatics streaming STT.
 *
 * @example
 * ```ts
 * import * as speechmatics from "getpatter/stt/speechmatics";
 * const stt = new speechmatics.STT();              // reads SPEECHMATICS_API_KEY
 * const stt = new speechmatics.STT({ apiKey: "sm_...", language: "en" });
 * ```
 */
export class STT extends _SpeechmaticsSTT {
  static readonly providerKey = "speechmatics";
  constructor(opts: SpeechmaticsSTTOptions = {}) {
    const key = opts.apiKey ?? process.env.SPEECHMATICS_API_KEY;
    if (!key) {
      throw new Error(
        "Speechmatics STT requires an apiKey. Pass { apiKey: 'sm_...' } or " +
          "set SPEECHMATICS_API_KEY in the environment.",
      );
    }
    super(key, opts);
  }
}

File: apps/patter-voice-sdk/libraries/python/getpatter/stt/speechmatics.py
"""Speechmatics streaming STT for Patter pipeline mode."""

from __future__ import annotations

import os

from getpatter.providers.speechmatics_stt import (
    SpeechmaticsSTT as _SpeechmaticsSTT,
    TurnDetectionMode,
)

__all__ = ["STT", "TurnDetectionMode"]


class STT(_SpeechmaticsSTT):
    """Speechmatics real-time streaming STT.

    Example::

        from getpatter.stt import speechmatics

        stt = speechmatics.STT()            # reads SPEECHMATICS_API_KEY
        stt = speechmatics.STT(api_key="...", language="en")
    """

    def __init__(
        self,
        api_key: str | None = None,
        *,
        language: str = "en",
        turn_detection_mode: TurnDetectionMode = TurnDetectionMode.ADAPTIVE,
        sample_rate: int = 16000,
        enable_diarization: bool = False,
        include_partials: bool = True,
    ) -> None:
        key = api_key or os.environ.get("SPEECHMATICS_API_KEY")
        if not key:
            raise ValueError(
                "Speechmatics STT requires an api_key. Pass api_key='...' or "
                "set SPEECHMATICS_API_KEY in the environment."
            )
        super().__init__(
            api_key=key,
            language=language,
            turn_detection_mode=turn_detection_mode,
            sample_rate=sample_rate,
            enable_diarization=enable_diarization,
            include_partials=include_partials,
        )
